gateway/gateway.go

```diff
@@ -922,24 +922,27 @@ func (gateway *HandleT) pendingEventsHandler(w http.ResponseWriter, r *http.Requ
 		}
 	}
 
+	ctx, cancel := context.WithTimeout(r.Context(), config.GetDuration("Gateway.pendingEventsQueryTimeout", time.Duration(10), time.Second))
+	defer cancel()
+
 	var pending bool
 	if !excludeGateway {
-		pending = gateway.readonlyGatewayDB.HavePendingJobs([]string{CustomVal}, -1, gwParameterFilters)
-		if pending {
-			w.Write([]byte(fmt.Sprintf("{ \"pending_events\": %d }", getIntResponseFromBool(pending))))
+		pending, err = gateway.readonlyGatewayDB.HavePendingJobs(ctx, []string{CustomVal}, -1, gwParameterFilters)
+		if err != nil || pending {
+			w.Write([]byte(fmt.Sprintf("{ \"pending_events\": %d }", getIntResponseFromBool(true))))
 			return
 		}
 	}
 
-	pending = gateway.readonlyRouterDB.HavePendingJobs(nil, -1, rtParameterFilters)
-	if pending {
-		w.Write([]byte(fmt.Sprintf("{ \"pending_events\": %d }", getIntResponseFromBool(pending))))
+	pending, err = gateway.readonlyRouterDB.HavePendingJobs(ctx, nil, -1, rtParameterFilters)
+	if err != nil || pending {
+		w.Write([]byte(fmt.Sprintf("{ \"pending_events\": %d }", getIntResponseFromBool(true))))
 		return
 	}
 
-	pending = gateway.readonlyBatchRouterDB.HavePendingJobs(nil, -1, rtParameterFilters)
-	if pending {
-		w.Write([]byte(fmt.Sprintf("{ \"pending_events\": %d }", getIntResponseFromBool(pending))))
+	pending, err = gateway.readonlyBatchRouterDB.HavePendingJobs(ctx, nil, -1, rtParameterFilters)
+	if err != nil || pending {
+		w.Write([]byte(fmt.Sprintf("{ \"pending_events\": %d }", getIntResponseFromBool(true))))
 		return
 	}
 
```

Comment:
Looking at the implementation of `getWarehousePending`, we return `false` in case of an `error` which appears to be in contrast with what we are doing elsewhere. Is this intentional? Any idea why?

Reply:
Not sure why. But I understand there is a contrast.

One thing is getting warehouse status is not db intensive. So, I suggest for now leave it as is. We can have warehouse team address this later.